### PR TITLE
UISEES-33: Replacing existing query search

### PR DESCRIPTION
**Pull request by Dmytro-Melnyshyn**
image

PR for the [sub-task](https://issues.folio.org/browse/UISEES-37).

The purpose is to replace the `Query search` mode with an `Advanced search`. When the user selects `Advanced search` mode a new `ElasticQueryField` component appears.

For this, I need to work with a `SearchAndSort` stripes-smart-component and a `SearchField` stripes-component. To only work in an ui-inventory-es module (POC) I copied these components into the module.
I created new a `ElasticQueryField` component ([on initial stage](https://github.com/folio-org/ui-inventory-es/pull/2/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md#diff-81765dee44697d24c12cf957c862f8e88da01712a678edc61341ec2b218c499e)), added `isAdvancedSearch` and `searchButtonRef` props to the `SearhcField` component ([lines](https://github.com/folio-org/ui-inventory-es/pull/2/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md#diff-2f4083a1a4881b81a75ed2c72be2c7f1dd797713ad3c0c4854823dfbf6269e5c) 1020, 1025), expanded the `SearchField` component with the `ElasticQueryField` component ([lines](https://github.com/folio-org/ui-inventory-es/pull/2/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md#diff-2f4083a1a4881b81a75ed2c72be2c7f1dd797713ad3c0c4854823dfbf6269e5c) 114 - 140), added the description to a [storybook](https://github.com/folio-org/ui-inventory-es/pull/2/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md#diff-29424a9479063ba66151a6be35ef4fab1c942c26ca9a69bbdd8060383c3314ce), renamed `Query Search` to `Advanced search` ([1](https://github.com/folio-org/ui-inventory-es/pull/2/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md#diff-29424a9479063ba66151a6be35ef4fab1c942c26ca9a69bbdd8060383c3314ce), [2](https://github.com/folio-org/ui-inventory-es/pull/2/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md#diff-03423e533e7101fe103a71bd392486b92db5623a15d76ce0bbd3a96c39668017), [3](https://github.com/folio-org/ui-inventory-es/pull/2/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md#diff-31a9388ff2727472991ede393aeda2a3e3aa338f12e3ca8533f523973852223a), [4](https://github.com/folio-org/ui-inventory-es/pull/2/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md#diff-fe3175200f90ffae7775f7577709846f57016b2e56c29a52c6821a2000fd47d3), [5](https://github.com/folio-org/ui-inventory-es/pull/2/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.md#diff-df86c409be71a694cc9b8d7354838283f18c66b82ab9714c5a8ec19676619bc9)).

### CODE REVIEW

File: src/components/ElasticQueryField/ElasticQueryField.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import TextArea from '@folio/stripes-components/lib/TextArea';
+
+const defaultStyle = {
```
Comment (Godlevskyi) [inline src/components/ElasticQueryField/ElasticQueryField.js]: Please remove inline styles. I believe currents TextArea styles are ok for POC. If in future we would need to change it I belive we would need to add the ability for TextArea to extend base styles.

Comment (Dmytro-Melnyshyn) [reply src/components/ElasticQueryField/ElasticQueryField.js]: Thanks for the feedback, fixed.

**Review by zburke — APPROVED** [review]
I'm uncertain what kind of feedback you want from me here. 

It looks like this is an initial PR to pull a bunch of core components into your local project, as discussed. From that point of view it looks good. 

I'm a watcher on this repo so I get notifications about the activity here. If there are specific things you want feedback on, feel free to tag me for review, but otherwise don't worry about it. I don't want to slow you down!

**Review by Godlevskyi — APPROVED** [review]

**Comment by Dmytro-Melnyshyn** [comment]
> I'm uncertain what kind of feedback you want from me here.
> 
> It looks like this is an initial PR to pull a bunch of core components into your local project, as discussed. From that point of view it looks good.
> 
> I'm a watcher on this repo so I get notifications about the activity here. If there are specific things you want feedback on, feel free to tag me for review, but otherwise don't worry about it. I don't want to slow you down!

@zburke Thanks for the review, I agree with you.

**Pull request merged**